src/panels/lovelace/cards/hui-entity-button-card.ts

```diff
@@ -62,7 +62,12 @@ class HuiEntityButtonCard extends LitElement implements LovelaceCard {
       throw new Error("Invalid Entity");
     }
 
-    this._config = { theme: "default", ...config };
+    this._config = {
+      theme: "default",
+      tap_action: { action: "toggle" },
```

Comment:
Okay, I should have put my comment here 😆

Reply:
Something like this?

```typescript
this._config = {
      theme: "default",
      tap_action: {
        action: DOMAINS_TOGGLE.has(config.entity.split(".", 1)[0])
          ? "toggle"
          : "more-info",
      },
      hold_action: { action: "more-info" },
      ...config,
    };
```

Reply:
Is my linter drunk...?
```
Type '{ entity: string; name?: string | undefined; icon?: string | undefined; theme: string; tap_action: ToggleActionConfig | CallServiceActionConfig | NavigateActionConfig | MoreInfoActionConfig | NoActionConfig | { ...; }; hold_action: ToggleActionConfig | ... 4 more ... | { ...; }; index?: number | undefined; view_inde...' is not assignable to type 'Config'.
  Types of property 'tap_action' are incompatible.
    Type 'ToggleActionConfig | CallServiceActionConfig | NavigateActionConfig | MoreInfoActionConfig | NoActionConfig | { action: "toggle" | "more-info"; }' is not assignable to type 'ToggleActionConfig | CallServiceActionConfig | NavigateActionConfig | MoreInfoActionConfig | NoActionConfig | undefined'.
      Type '{ action: "toggle" | "more-info"; }' is not assignable to type 'ToggleActionConfig | CallServiceActionConfig | NavigateActionConfig | MoreInfoActionConfig | NoActionConfig | undefined'.
        Type '{ action: "toggle" | "more-info"; }' is not assignable to type 'NoActionConfig'.
          Types of property 'action' are incompatible.
            Type '"toggle" | "more-info"' is not assignable to type '"none"'.
              Type '"toggle"' is not assignable to type '"none"'.ts(2322)
(property) HuiEntityButtonCard._config?: Config | undefined
```

Reply:
Ha, this is a bug in typescript. It is trying to map `action: "toggle" | "more-info"` and trying to find a single type that matches it, instead of realizing that we're referencing  `ToggleActionConfig` and `MoreInfoActionConfig`.

You will need to pull it out into an `if` statement. Probably good too, because we wouldn't want hold_action to be `more-info` if `tap` is `more-info`. 

Also, use `computeDomain`.